ckanext/xloader/jobs.py

```diff
@@ -479,23 +478,41 @@ def update_resource(resource, patch_only=False):
     or patch the given CKAN resource for file hash
     """
     action = 'resource_update' if not patch_only else 'resource_patch'
-    from ckan import model
-    user = get_action('get_site_user')({'model': model, 'ignore_auth': True}, {})
-    context = {'model': model, 'session': model.Session, 'ignore_auth': True,
-               'user': user['name'], 'auth_user_obj': None}
+    user = get_action('get_site_user')({'ignore_auth': True}, {})
+    context = {
+        'ignore_auth': True,
+        'user': user['name'],
+        'auth_user_obj': None
+    }
     get_action(action)(context, resource)
 
 
+def _get_user_from_key(api_key_or_token):
+    """ Gets the user using the API Token or API Key.
+
+    This method provides backwards compatibility for CKAN 2.9 that
+    supported both methods and previous CKAN versions supporting
+    only API Keys.
+    """
+    user = None
+    if get_user_from_token:
+        user = get_user_from_token(api_key_or_token)
+    if not user:
+        user = model.Session.query(model.User).filter_by(
+            apikey=api_key_or_token
```

Comment:
@ThrawnCA : I believe CKAN security is improving which is why this is now in place.

as well as on line 512 etc.